reference_test.go

```diff
@@ -117,7 +118,7 @@ func TestReferenceParse(t *testing.T) {
 			tag:        "Uppercase",
 		},
 		{
-			input: strings.Repeat("a/", 128) + "a:tag",
+			input: "domain/" + strings.Repeat("a", 256) + ":tag",
```

Comment:
This example may be incorrect here, because the first component is only considered to be a domain if it either contains a period (`.`), a colon (`:`) or it's literally `localhost` https://github.com/distribution/reference/blob/8507c7fcf0da9f570540c958ea7b972c30eeaeca/reference.go#L8-L10


If we intend to have a domain here, it should probably contain a `.`, and use one of the designated [RFC 6761]()https://www.rfc-editor.org/rfc/rfc6761.html) domains for this (e.g., `example.com` or `example.org`, or [`<something>.test`](https://www.rfc-editor.org/rfc/rfc6761.html#section-6.2)